### DISCUSSION TITLE
Suggestion: Sub-items

### DISCUSSION THREAD

**Discussion by DLvalentine** [jotaen/xit#60] [File Format]
Sometimes when working on more complex tasks, I'll have items that I could easily break out into sub-items. For now, I've been solving that with some clever grouping or tags (e.g. #dependsOn #featureA)  - I think it could be useful to consider indentation valid, or to have some sort of characters baked in to tags to make it more clear when I have a task that should be completed before another task. 

Sounds silly, of course, since there's also the priority (!). With any combination of priority, tags, and groups, I could probably make it clear within the file itself, but having a structure provided by the format might be useful. Just a thought!

**Comment by jotaen** [comment]
Thanks for your sharing your thoughts here. The idea of using tags for expressing dependencies didn’t occur to me yet, that’s a pretty interesting approach – although it might also feel a bit “technical”. Tag values might come in handy here, like:

```
[ ] Do #abc
[ ] Do that #before=abc
[ ] Do that #before=abc
```

There is a similar [discussion about nested items](https://github.com/jotaen/xit/discussions/2), which proposes this syntax:

```
[ ] Task
    [ ] Subtask 1
    [ ] Subtask 2
```

The discussion thread is pretty lengthy – the tldr is basically that such a syntax is certainly valid, intuitive, and useful, but it has some complexity to it in regards to the specification, and it might not be possible to implement “correctly” in certain tooling, e.g. in many (most?) syntax highlighting engines. I find the latter point to be a major issue, unfortunately, because [x]it! files are supposed to be edited by hand, so I think the editor should give a definite and correct answer as to whether the file is formatted correctly or not.
